### ISSUE TITLE
Safari response error node-fetch

### ISSUE THREAD

**Issue by leimonio** [kadirahq/lokka-transport-http#13]
On each response received on Safari (v9.1.1) the following error occurs:
`TypeError: Error.captureStackTrace is not a function. (In 'Error.captureStackTrace(this, this.constructor)', 'Error.captureStackTrace' is undefined)`

The response of the request is completed, I can see in Network DevTools the response of the server, but the above error crashes the data loading on page.

This error is coming from a dependency of `lokka-transport-http`, the `node-fetch` package.

Note: I have tried both versions on `lokka-transport-http`, I happend only in `1.4.0` version (latest), not in `1.3.1` (available in npm).

**Comment by mbaroukh** [comment]
I have the same problem today. This is a problem but it is not the only problem. It happen because there is an exception.
But the real problem behind, for me, is gzip compression. When disabled, it works.

It happen only for graphql queries, only with Lokka and only on safary (and so iOS ...)

**Comment by leimonio** [comment]
@mbaroukh I have solved this issue by downgrading the version to `1.3.1`. Though this is not the best way, but for now it works until this issue is resolved.

**Comment by mbaroukh** [comment]
Thanks for your hint. I'll try.
Are you using React ?
Because here :
https://github.com/kadirahq/lokka-transport-http/blob/master/lib/index.js

The test to use node-fetch is :

else if (typeof __dirname !== 'undefined') {
  // for Node.js
  fetchUrl = require('node-fetch');

And on my app with react,  (typeof __dirname !== 'undefined') return true ...

**Comment by leimonio** [comment]
It's been a while but I think that was also the issue I had to face. In the latest version `isomorphic-fetch` (used to exist in `1.3.3`) was removed in favor of `whatwg-fetch` and `node-fetch`. `whatwg-fetch` is supposed to be used for the client side requests, while the `node-fetch` for the server side ones, apparently. But not sure why `typeof __dirname !== 'undefined'` returns `true` on client.

EDIT:
When using webpack this condition `typeof __dirname !== 'undefined'` comes compiled to `true`.
